profile/profile_test.go

```diff
@@ -811,7 +811,7 @@ func locationHash(s *Sample) string {
 	return tb
 }
 
-func TestInferUnits(t *testing.T) {
+func TestNumLabelUnits(t *testing.T) {
```

Comment:
Do the modified tests fail with the original code in profile.go for the reason same as in the issue? I am not sure they do. I actually just tried it, here is what I see:

```
profile_test.go:925: One sample, one key with multiple values and three different units: for key key1, got unit key1, want unit bytes
```

But this failure is unrelated - this is from the second case you've added which tests that if there are more units than there unit values, we still process the units for the values rather than pretending there are no units at all. This change is unrelated to the issue I believe?

I believe the test case that is missing is something like modifying this existing one:

```
                {
                        "Key alignment, unit not specified",
-                       []map[string][]int64{{"alignment": {8}}},
-                       []map[string][]string{nil},
+                       []map[string][]int64{{"alignment": {8}}, {"alignment": {8}}},
+                       []map[string][]string{nil, nil},
                        map[string]string{"alignment": "bytes"},
                        map[string][]string{},
                },
```

Plus, (and probably more importantly), the comparison of want and got maps for units and ignoredUnits does not quite test the equality and needs to be updated. These are now done as `for key, wantUnit := range test.wantUnits {` and `for key, ignored := range ignoredUnits` respectively and they don't seem to compare the size of want and got maps before the loop, so these only seem to check that test.wantUnits is a subset of gotUnits and that gotIgnoredUnits is a subset of test.wantIgnoredUnits. Besides the size checks, make what you iterate by consistent perhaps - there isn't a good reason to have the loop over want things in the first loop but by got things in the second loop, is there?

Reply:
Modified tests did failed, but I added the test you described. And I modified comparison of want and got for this test.

Reply:
FWIW, yes, the tests failed with the error I mentioned, but as I said I think they failed for reason unrelated to the one that caused the warning messages I think.